src/clusterfuzz/_internal/bot/tasks/trials.py

```diff
@@ -92,9 +85,14 @@ def setup_additional_args_for_app(self, shuffle=True):
       random.shuffle(trial_keys)
 
     for app_args in trial_keys:
-      if (random.random() < self.trials[app_args].probability and
-          self.doesnt_asymmetrically_contradicts(app_args, contradicts,
-                                                 trial_args)):
+      if random.random() < self.trials[app_args].probability:
+        # Check if the flag is contradicted by an already added flag
+        if app_args in contradicts:
+          continue
+        # Check if the flag contradicts an already added flag
+        if any(
+            flag in self.trials[app_args].contradicts for flag in trial_args):
```

Comment:
The optimization you added before vanished now?

if self.trials[app_args].contradicts and any(...

Reply:
done